runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -9833,7 +9846,7 @@ static TR::Register *inlineConcurrentLinkedQueueTMOffer(TR::Node *node, TR::Code
 
       VMnonNullSrcWrtBarCardCheckEvaluator(node, nReg, pReg, cndReg, qReg, retryCountReg, temp3Reg, temp4Reg, wrtbar1Donelabel, conditions, false, cg, false);
       // ALG: *** wrtbar1Donelabel
-      generateLabelInstruction(cg, TR::InstOpCode::label, node, wrtbar1Donelabel);
+      generateDepLabelInstruction(cg, TR::InstOpCode::label, node, wrtbar1Donelabel, conditions);
```

Comment:
This looks like the right thing to do, I'm just wondering how the existing code survived this long without it. There's clearly internal control flow going on in this routine without even considering what `VMnonNullSrcWrtBarCardCheckEvaluator` does.

Putting on my technical debt reduction hat for a minute, I _think_ it's because this code only runs when transactional memory is available on the processor, enabled in the JVM/JIT, and then only with a specific version of the `ConcurrentLinkedQueue` class (that has extra methods related to TM) that doesn't exist in OpenJ9. Long story short I think this code is dead and we should delete it. @fjeremic is also working on changes to how we do register dependencies to free us from having to count how many we deps we need so hopefully we can clean all of this up one day soon.

Reply:
In this case is the dead code the entire `inlineConcurrentLinkedQueueTMOffer()` function?